core/src/test/java/com/google/errorprone/bugpatterns/MissingOverrideTest.java

```diff
@@ -89,4 +90,32 @@ public void deprecatedMethod() throws Exception {
             "Test.java", "public class Test extends Super {", "  public void f() {}", "}")
         .doTest();
   }
+
+  @Test
+  public void interfaceOverride() throws Exception {
+    compilationHelper
+        .addSourceLines("Super.java", "interface Super {", "  void f();", "}")
+        .addSourceLines(
+            "Test.java",
+            "public interface Test extends Super {",
+            "  // BUG: Diagnostic contains: f implements method in Super; expected @Override",
+            "  void f();",
+            "}")
+        .doTest();
+  }
+
+  // this doesn't work!  need new APIs in CompilationTestHelper
```

Comment:
Would appreciate some advice on how to test.  The most obvious route I could was to add APIs to `CompilationTestHelper` to allow for passing in custom `ErrorProneFlags`.  But I wanted to check if there is an easier way before going ahead with that.

Reply:
Hmm, why doesn't this work? In some private code I do the following, which does work:
```java
@RunWith(JUnit4.class)
public final class LexicographicalAnnotationAttributeListingCheckTest {
  ...
  private final CompilationTestHelper restrictedCompilationTestHelper =
      CompilationTestHelper.newInstance(
              LexicographicalAnnotationAttributeListingCheck.class, getClass())
          .setArgs(
              ImmutableList.of(
                  "-XepOpt:LexicographicalAnnotationAttributeListing:Includes=pkg.A.Foo,pkg.A.Bar",
                  "-XepOpt:LexicographicalAnnotationAttributeListing:Excludes=pkg.A.Bar#value"));
  ...
  @Test
  public void testFiltering() {
    restrictedCompilationTestHelper
        .addSourceLines(...)
        .doTest();
  }
}
```